common/persistence/client/quotas.go

```diff
@@ -74,6 +76,38 @@ var (
 )
 
 func NewPriorityRateLimiter(
+	namespaceMaxQPS PersistenceNamespaceMaxQps,
+	hostMaxQPS PersistenceMaxQps,
+	perShardNamespaceMaxQPS PersistencePerShardNamespaceMaxQPS,
+	requestPriorityFn quotas.RequestPriorityFn,
+) quotas.RequestRateLimiter {
+	return quotas.NewMultiRequestRateLimiter(
+		newPriorityNamespaceRateLimiter(namespaceMaxQPS, hostMaxQPS, requestPriorityFn),
+		newPerShardPerNamespaceRateLimiter(perShardNamespaceMaxQPS),
+	)
```

Comment:
- consume per shard per ns rl first. I understand multi-rl implementation will reserve and cancel the reservation if tokens can be get from later rate limiters, but in practice it doesn't work that well. Keep reserving & cancelling will result in effectively a lower rps and in extreme case no request is able to go through.

- Is it possible to do some refactoring here and simplify the rate limiter structure here a little bit?